### ISSUE TITLE
[Hooks] - Migration guide from unofficial apollo-hooks

### ISSUE THREAD

**Issue by wendelfreitas** [apollographql/react-apollo#3328]
What do you think about having a migration guide to help you migrate from unofficial to apollo hooks.

I had and am having a lot of problems because I am migrating an app that used its unofficial hooks throughout its structure and still haven't solved 90% of the problems.

**Comment by hwillson** [comment]
@wendelfreitas If anyone is interested in putting together a migration guide, we'd be happy to host it. Question though - what problems are you encountering?

**Comment by jinshin1013** [comment]
I have migrated from `react-apollo-hooks` to `react-apollo@v3` and haven't yet encountered any problems except some Typescript interface name difference but it isn't really a problem.

**Comment by wendelfreitas** [comment]
@hwillson 
I have problems with mutations, since in react-apollo-hooks we had no error handling and no loading, so I made a custom hook to solve and then I migrated to the official hooks and then I started to have network errors. , example: Error: Network error: null is not an object (evaluation _ref2.token).

@jinshin1013  Good job bro, I do not say problems with the tool itself, but in the migration process, perhaps, it was just me with the difficulty.
